lib/engines/hcktest/hcktest.rb

```diff
@@ -225,19 +226,64 @@ def prepare_tests
         return
       end
 
-      @tests.list_tests(log: true)
+      @test_list = @tests.list_tests(log: true)
     end
 
     def auto_run
       ResourceScope.open do |scope|
-        run_and_configure_setup scope
-        prepare_tests
+        run_studio scope
+        sleep 5 until @studio.up?
+
+        run_tests_without_config
+        run_tests_with_config
+
+        @tests.create_project_package
+      end
+    end
 
-        @tests.run
+    def run_tests_without_config
+      ResourceScope.open do |scope|
+        run_clients_and_configure_setup scope
+        prepare_tests
 
+        @logger.info('Environment ready, running normal tests')
+        @tests.run(group_tests[:normal])
         pause_run if @project.options.test.manual
+      end
+    end
 
-        @tests.create_project_package
+    def run_tests_with_config
+      group_tests.each do |group, tests|
+        next if tests.empty? || group == :normal
+
+        ResourceScope.open do |scope|
+          run_clients_and_configure_setup(scope, client_run_options(group))
+
+          @tests.run(tests)
+
+          pause_run if @project.options.test.manual
+          break if @project.run_terminated
+        end
+      end
+    end
+
+    def group_tests
+      grouped_tests = { normal: [], secure: [] }
+
+      @config.tests_config.each do |test_group|
+        selected_tests = @test_list.select { |test| test_group.tests.include?(test['name']) }
+        grouped_tests[:secure] += selected_tests if test_group.secure
+      end
+
+      grouped_tests[:normal] += (@test_list - grouped_tests.values.flatten)
+      grouped_tests
+    end
```

Comment:
Let's split this into a method that returns tests with config and another that returns tests without config.